src/layer.rs

```diff
@@ -128,13 +128,24 @@ where
 
     fn validate_config(&self) -> Result<(), EtwError> {
         #[cfg(target_os = "linux")]
-        if self
-            .provider_name
-            .contains(|f: char| !f.is_ascii_alphanumeric())
         {
-            // The perf command is very particular about the provider names it accepts.
-            // The Linux kernel itself cares less, and other event consumers should also presumably not need this check.
-            return Err(EtwError::InvalidProviderNameCharacters(self.provider_name.clone()));
+            if self
+                .provider_name
+                .contains(|f: char| !f.is_ascii_alphanumeric() && f != '_')
+            {
+                // The perf command is very particular about the provider names it accepts.
+                // The Linux kernel itself cares less, and other event consumers should also presumably not need this check.
+                return Err(EtwError::InvalidProviderNameCharacters(self.provider_name.clone()));
+            }
+
+            let group_name_len = match &self.provider_group {
+                None => 0,
+                Some(ref name) => Mode::get_provider_group(&name).as_ref().len()
```

Comment:
I'm OK with saying len() is the number of characters, since we require it to be ASCII characters elsewhere.

Reply:
str.len() returns the number of bytes, which is perfectly acceptable here.

https://doc.rust-lang.org/std/primitive.str.html#method.len